src/java/org/apache/cassandra/utils/Throwables.java

```diff
@@ -230,6 +230,24 @@ public static RuntimeException unchecked(Throwable t)
         return t instanceof RuntimeException ? (RuntimeException)t : new RuntimeException(t);
     }
 
+    /**
+     * throw the exception without wrapping as a runtime exception.  This relies on a trick to stop the compiler from
+     * forcing checking of checked exceptions so a exception can be rethrown unchecked
+     */
+    public static RuntimeException throwAsUncheckedException(Throwable t)
```

Comment:
This is based on something that we cannot guaranty will work in the future and I am not truly convinced of its benefit over wrapping the Exception in a Runtime one.